crates/rpc/src/v03/method/get_state_update.rs

```diff
@@ -13,39 +13,33 @@ pub async fn get_state_update(
     context: RpcContext,
     input: GetStateUpdateInput,
 ) -> Result<types::StateUpdate, GetStateUpdateError> {
-    let block_id = match input.block_id {
-        BlockId::Pending => {
-            match &context
-                .pending_data
-                .ok_or_else(|| anyhow!("Pending data not supported in this configuration"))?
-                .state_update()
-                .await
-            {
-                Some(update) => {
-                    let update = update.as_ref().clone().into();
-                    return Ok(update);
-                }
-                None => return Err(GetStateUpdateError::BlockNotFound),
-            }
-        }
-        other => other.try_into().expect("Only pending cast should fail"),
-    };
-
-    let storage = context.storage.clone();
     let span = tracing::Span::current();
 
-    let jh = tokio::task::spawn_blocking(move || {
+    tokio::task::spawn_blocking(move || {
         let _g = span.enter();
-        let mut db = storage
+        let mut db = context
+            .storage
             .connection()
             .context("Opening database connection")?;
 
         let tx = db.transaction().context("Creating database transaction")?;
 
-        get_state_update_from_storage(&tx, block_id)
-    });
+        let state_update = match input.block_id {
+            BlockId::Pending => context
+                .pending_state_update(&tx)
+                .context("Querying pending state update")?
+                .ok_or(GetStateUpdateError::BlockNotFound)?
```

Comment:
I'm unsure if these shouldn't just return an empty state update or not. But this was the current behaviour.